File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int[] right=new int[prices.length];
        if(prices.length==1)
        {
            return 0;
        }
        right[prices.length-1]=prices[prices.length-1];
        for(int i=prices.length-2;i>=0;i--)
        {
            right[i]=Math.max(right[i+1],prices[i+1]);
        }

        int max=0;
        for(int i=0;i<prices.length;i++)
        {
            if(right[i]-prices[i]>max)
            {
                max=right[i]-prices[i];
            }
        }

        return max;

    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def time(self,k:int,piles: List[int])->k:
            sum=0
            for i in piles:
                sum+=math.ceil(i/k)
            
            return sum
    
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l=1
        r=max(piles)
        minimum=r

        while(l<=r):
            m=(l+r)//2
            t=self.time(m,piles)
            if t<=h:
                minimum=m
                r=m-1
            else:
                l=m+1
        
        return minimum




        
        
